presto-main/src/main/java/com/facebook/presto/metadata/MetadataManager.java

```diff
@@ -141,20 +141,20 @@ public MetadataManager(FeaturesConfig featuresConfig,
             AnalyzePropertyManager analyzePropertyManager,
             TransactionManager transactionManager)
     {
-        this(featuresConfig,
-                typeManager,
+        this(typeManager,
                 createTestingViewCodec(),
                 blockEncodingSerde,
                 sessionPropertyManager,
                 schemaPropertyManager,
                 tablePropertyManager,
                 columnPropertyManager,
                 analyzePropertyManager,
-                transactionManager);
+                transactionManager,
+                new FunctionManager(typeManager, blockEncodingSerde, featuresConfig));
```

Comment:
kinda weird for an injected class. The constructor `MetadataManager` is only used for testing purpose. Maybe add `VisibleForTesting`.

Reply:
I added that, then realized that some of the call site are borderline "testing". For example `LocalQueryRunner`.